tests/Shipment/TaxIdentificationNumbers/TaxIdentificationNumberCollectionTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Shipment\TaxIdentificationNumbers;
+
+use Faker\Factory;
+use Mockery;
+use MyParcelCom\Integration\Shipment\Items\Item;
+use MyParcelCom\Integration\Shipment\Items\ItemCollection;
+use MyParcelCom\Integration\Shipment\TaxIdentificationNumbers\Enums\TaxNumberTypeEnum;
+use MyParcelCom\Integration\Shipment\TaxIdentificationNumbers\TaxIdentificationNumber;
+use MyParcelCom\Integration\Shipment\TaxIdentificationNumbers\TaxIdentificationNumberCollection;
+use PHPUnit\Framework\TestCase;
+use function random_int;
+
+class TaxIdentificationNumberCollectionTest extends TestCase
+{
+    public function test_it_should_convert_items_collection_to_array(): void
+    {
+        $faker = Factory::create();
+        $firstNumber = Mockery::mock(TaxIdentificationNumber::class, [
+            'toArray' => [
+                'country_code' => $faker->countryCode,
+                'type'         => TaxNumberTypeEnum::EORI(),
+                'description'  => $faker->text(25),
+                'number'       => $faker->text(9),
+            ],
+        ]);
+        $secondNumber = Mockery::mock(TaxIdentificationNumber::class, [
+            'toArray' => [
+                'country_code' => $faker->countryCode,
+                'type'         => TaxNumberTypeEnum::IOSS(),
+                'description'  => $faker->text(25),
+                'number'       => $faker->text(9),
+            ],
+        ]);
+
+        $collection = new TaxIdentificationNumberCollection(
+            $firstNumber,
+            $secondNumber
+        );
+
+        self::assertEquals([$firstNumber->toArray(), $secondNumber->toArray()], $collection->toArray());
```

Comment:
With the conversation `toArray()` in the expected we involve logic within the test instead of relying on static expected values. Basically, the invocation of `toArray` on each item makes the expectation dynamic. The problem with is that then the test become sensitive to that logic, while it should have pinned expectations. 

I'd suggest that if you still want to use faker (which is really cool btw) to first define the expected arrays as separate variables, then use them on the two different contexts - one to feed the mocks and one to feed the expectations. That way the test does not rely on calling the toArray() on items themselves.

Reply:
I was thinking about doing exactly this because I understood the dynamic possibilities on the function call. I will change this.